static/js/popovers.js

```diff
@@ -1050,6 +1198,80 @@ exports.register_click_handlers = function () {
             last_scroll = date;
         });
     }());
+
+    $("body").on("click", "#send_later", function (e) {
+        e.preventDefault();
+        e.stopPropagation();
+
+        popovers.toggle_send_later_popover(this);
+    });
+
+    $("body").on("click", ".send_later_hours", function (e) {
+        var send_later_in = $(e.currentTarget).attr('id');
+        var send_at_time = moment().add(send_later_hours[send_later_in].hours, 'h');
+        var message_content = compose_state.message_content();
+        var command = "/schedule " + send_at_time.format("MMM D YYYY h:mm a") + "\n";
+
+        if (reminder.is_deferred_delivery(message_content)) {
+            var raw_message = message_content.split('\n');
+            message_content = raw_message.slice(1).join('\n');
+        }
+
+        message_content = command + message_content;
+        compose_state.message_content(message_content);
+        popovers.hide_send_later_popover();
+        e.stopPropagation();
+        e.preventDefault();
+    });
+
+    $("body").on("click", ".send_later_tomorrow", function (e) {
+        var send_later_at = $(e.currentTarget).attr('id');
+        var send_time = send_later_tomorrow[send_later_at].time;
+        var date = new Date();
+        var scheduled_date = date.setDate(date.getDate() + 1);
+        var send_at_time = moment(scheduled_date).format("MMM D YYYY ") + send_time;
+        var message_content = compose_state.message_content();
+        var command = "/schedule " + send_at_time + "\n";
+
+        if (reminder.is_deferred_delivery(message_content)) {
+            var raw_message = message_content.split('\n');
+            message_content = raw_message.slice(1).join('\n');
+        }
+
+        message_content = command + message_content;
+        compose_state.message_content(message_content);
+        popovers.hide_send_later_popover();
+        e.stopPropagation();
+        e.preventDefault();
+    });
+
+    $("body").on("click", ".send_later_days_and_weeks", function (e) {
+        var send_later_in = $(e.currentTarget).attr('id');
+        var send_at_time = moment().add(send_later_days_and_weeks[send_later_in].days, 'd');
+        var message_content = compose_state.message_content();
+        var command = "/schedule " + send_at_time.format("MMM D YYYY h:mm a") + "\n";
+
+        if (reminder.is_deferred_delivery(message_content)) {
+            var raw_message = message_content.split('\n');
+            message_content = raw_message.slice(1).join('\n');
+        }
+
+        message_content = command + message_content;
+        compose_state.message_content(message_content);
+        popovers.hide_send_later_popover();
+        e.stopPropagation();
+        e.preventDefault();
+    });
```

Comment:
So, all 3 of the above functions have essentially the same content.  I think it makes sense to extract the common code that comes after computing `send_at_time` into a function, which maybe belongs in a different module, with a name like `compose_actions.set_compose_box_schedule(send_at_time)`.

That would eliminate a lot of code duplication (and risk of resulting bugs), and also probably be more readable.